src/runtime_src/xdp/profile/database/static_info_database.cpp

```diff
@@ -1428,57 +1403,213 @@ namespace xdp {
     }
+
     {
       std::lock_guard<std::mutex> lock(summaryLock) ;
       systemDiagram = buf.str() ;
     }
+  }
+
+  void VPStaticDatabase::addPortInfo(XclbinInfo* currentXclbin,
+                                     const char* systemMetadataSection,
+                                     size_t systemMetadataSz)
+  {
+    if (currentXclbin == nullptr || systemMetadataSection == nullptr ||
+        systemMetadataSz <= 0)
+      return;
 
+    // Parse the SYSTEM_METADATA section using boost property trees, which
+    // could throw exceptions in multiple ways.
     try {
       std::stringstream ss;
       ss.write(systemMetadataSection, systemMetadataSz);
 
-      // Create a property tree and determine if the variables are all default values
+      // Create a property tree based off of the JSON
       boost::property_tree::ptree pt;
       boost::property_tree::read_json(ss, pt);
 
-      currentXclbin->name = pt.get<std::string>("system_diagram_metadata.xclbin.generated_by.xclbin_name", "");
-      if(!currentXclbin->name.empty()) {
-        currentXclbin->name += ".xclbin";
+      auto top = pt.get_child("system_diagram_metadata");
+
+      // Parse the xsa section for memory topology information
+      auto xsa = top.get_child("xsa");
+      auto device_topology = xsa.get_child("device_topology");
+
+      // Parse the xclbin section for compute unit port information
+      auto xclbin = top.get_child("xclbin");
+      auto user_regions = xclbin.get_child("user_regions");
+
+      // Temp data structure to hold mappings
+      std::map<std::string, int32_t> argumentToMemoryIndex;
+
+      // We also need to know which argument goes to which memory
+      for (auto& region : user_regions) {
+        for (auto& connection : region.second.get_child("connectivity")) {
+          auto node1 = connection.second.get_child("node1");
+          auto node2 = connection.second.get_child("node2");
+
+          auto arg = node1.get<std::string>("arg_name");
+          auto id = node2.get<std::string>("id");
+
+          if (id != "" && arg != "")
+            argumentToMemoryIndex[arg] = std::stoi(id);
+        }
+      }
+
+      auto tolower = [](char c) { return static_cast<char>(std::tolower(c));};
+
+      // Now go through each of the kernels to determine the port information
+      for (auto& region : user_regions) {
+        for (auto& kernel : region.second.get_child("kernels")) {
+          auto kernelName = kernel.second.get<std::string>("name", "");
+          for (auto& port : kernel.second.get_child("ports")) {
+            auto portName = port.second.get<std::string>("name");
+            auto portType = port.second.get<std::string>("port_type");
+            if (portName == "S_AXI_CONTROL" || portType == "stream")
+              continue;
+
+            auto portWidth = port.second.get<std::string>("data_width");
+	    std::transform(portName.begin(), portName.end(), portName.begin(),
```

Comment:
looks misaligned. Mix of tabs and spaces?

Reply:
Changed these lines to spaces to align in the latest commit